scripts/installation/install-openkat.yml

```diff
@@ -0,0 +1,588 @@
+      dest: /tmp
+
+  # sudo apt install --no-install-recommends ./kat-*_amd64.deb ./xtdb-http-multinode_*_all.deb
+  - name: Install all needed packages
+    apt:
+      deb: "/tmp/{{ item }}"
+      install_recommends: false
+    loop:
+      - "kat-boefjes_{{ katrelease }}_amd64.deb"
+      - "kat-bytes_{{ katrelease }}_amd64.deb"
+      - "kat-keiko_{{ katrelease }}_amd64.deb"
+      - "kat-mula_{{ katrelease }}_amd64.deb"
+      - "kat-octopoes_{{ katrelease }}_amd64.deb"
+      - "kat-rocky_{{ katrelease }}_amd64.deb"
+      - "xtdb-http-multinode_{{ xtdbrelease }}_all.deb"
+
+  # Set up the databases
+  # OpenKAT needs three databases for its components. One for rocky, KAT-alogus and bytes. The following steps will guide you through the creation of these databases.
+  # If you will be running the database on the same machine as KAT, you can install Postgres:
+  # sudo apt install postgresql
+  - name: Install other needed packages
+    apt:
+      name: "{{ item }}"
+      state: present
+    loop:
+      - python3-psycopg2
+      - python3-pexpect
+      - postgresql
+      - postgresql-client
+      - rabbitmq-server
+      - sudo
+
+  # Rocky DB
+  # Generate a secure password for the Rocky database user, as an example we’ll use /dev/urandom:
+  # echo $(tr -dc A-Za-z0-9 < /dev/urandom | head -c 20)
+  - name: Generate random password for Rocky DB
+    set_fact:
+      # Turns out that special characters create problems.
+      #var: "{{ lookup('community.general.random_string', length=20, min_numeric=3, min_upper=3, min_special=3) }}"
+      rocky_db_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+
+  - name: Show password for debugging purposes (TODO comment these lines)
+    debug:
+      var: rocky_db_password
+
+  # To configure rocky to use this password, open /etc/kat/rocky.conf and fill in this password for ROCKY_DB_PASSWORD.
+  - name: Store password in /etc/kat/rocky.conf
+    lineinfile:
+      path: /etc/kat/rocky.conf
+      regexp: "ROCKY_DB_PASSWORD=.*"
+      line:  "ROCKY_DB_PASSWORD={{ rocky_db_password }}"
+
+  # sudo -u postgres createuser rocky -P
+  - name: Create Postgresql user rocky and grant privs
+    become: true
+    become_user: postgres
+    become_method: sudo
+    postgresql_user:
+      name: rocky
+      password: "{{ rocky_db_password }}"
+      role_attr_flags: NOSUPERUSER
+
+  # sudo -u postgres createdb rocky_db
+  # sudo -u postgres psql -c 'GRANT ALL ON SCHEMA public TO rocky;' rocky_db
+  - name: Create Postgresql database rocky_db
+    become: true
+    become_user: postgres
+    become_method: sudo
+    postgresql_db:
+      name: rocky_db
+      owner: rocky
+      encoding: UTF-8
+      template: template0
+
+
+  # Now use rocky-cli to initialize the database:
+  # sudo -u kat rocky-cli migrate
+  - name: initialize Rocky database
+    become: true
+    become_user: kat
+    become_method: sudo
+    shell:
+      cmd: rocky-cli migrate
```

Comment:
Ook hier krijg ik nog een stacktrace....

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/__main__.py", line 4, in <module>
    main(prog="alembic")
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/config.py", line 632, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/config.py", line 626, in main
    self.run_cmd(cfg, options)
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/config.py", line 603, in run_cmd
    fn(
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/command.py", line 385, in upgrade
    script.run_env()
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/script/base.py", line 582, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/util/pyfiles.py", line 94, in load_python_file
    module = load_module_py(module_id, path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/alembic/util/pyfiles.py", line 110, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/bytes/database/migrations/env.py", line 20, in <module>
    config.set_main_option("sqlalchemy.url", get_settings().db_uri)
                                             ^^^^^^^^^^^^^^
  File "/opt/venvs/kat-bytes/lib/python3.11/site-packages/bytes/config.py", line 146, in get_settings
    return Settings()
           ^^^^^^^^^^
  File "pydantic/env_settings.py", line 39, in pydantic.env_settings.BaseSettings.__init__
    super().__init__(
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 1 validation error for Settings
queue_uri
  ensure this value has at least 1 characters (type=value_error.any_str.min_length; limit_value=1)

Reply:
Nee, dat komt omdat QUEUE_URI beneden pas goed gezet wordt. Hetzelfde voor die andere stacktrace. De migraties moeten pas gedraaid worden wanneer dit in de config aangepast is. Sinds 1.11 checken we beter of de config goed is zodat je duidelijker ziet bij het opstarten van een service als er iets niet goed is. Ik denk dat dit in eerdere versies wel werkte omdat QUEUE_URI niet noodzakelijk is voor de migraties.